packages/core/src/modules/credentials/protocol/v2/__tests__/v2.ldproof.credentials.propose-offerED25519.test.ts

```diff
@@ -0,0 +1,540 @@
+      expect(offerCredentialExchangeRecord.state).toEqual(CredentialState.RequestSent)
+      expect(offerCredentialExchangeRecord.threadId).not.toBeNull()
+
+      testLogger.test('Faber waits for credential request from Alice')
+      await waitForCredentialRecord(faberAgent, {
+        threadId: aliceCredentialRecord.threadId,
+        state: CredentialState.RequestReceived,
+      })
+
+      testLogger.test('Faber sends credential to Alice')
+
+      const options: AcceptRequestOptions = {
+        credentialRecordId: faberCredentialRecord.id,
+        comment: 'V2 Indy Credential',
+      }
+      await faberAgent.credentials.acceptRequest(options)
+
+      testLogger.test('Alice waits for credential from Faber')
+      aliceCredentialRecord = await waitForCredentialRecord(aliceAgent, {
+        threadId: faberCredentialRecord.threadId,
+        state: CredentialState.CredentialReceived,
+      })
+
+      testLogger.test('Alice sends credential ack to Faber')
+      await aliceAgent.credentials.acceptCredential(aliceCredentialRecord.id)
+
+      testLogger.test('Faber waits for credential ack from Alice')
+      faberCredentialRecord = await waitForCredentialRecord(faberAgent, {
+        threadId: faberCredentialRecord.threadId,
+        state: CredentialState.Done,
+      })
+      expect(aliceCredentialRecord).toMatchObject({
+        type: CredentialExchangeRecord.type,
+        id: expect.any(String),
+        createdAt: expect.any(Date),
+        threadId: expect.any(String),
+        connectionId: expect.any(String),
+        state: CredentialState.CredentialReceived,
+      })
+
+      const credentialMessage = await didCommMessageRepository.findAgentMessage({
+        associatedRecordId: faberCredentialRecord.id,
+        messageClass: V2IssueCredentialMessage,
+      })
+
+      expect(JsonTransformer.toJSON(credentialMessage)).toMatchObject({
+        '@type': 'https://didcomm.org/issue-credential/2.0/issue-credential',
+        '@id': expect.any(String),
+        comment: 'V2 Indy Credential',
+        formats: [
+          {
+            attach_id: expect.any(String),
+            format: 'aries/ld-proof-vc@1.0',
+          },
+        ],
+        'credentials~attach': [
+          {
+            '@id': expect.any(String),
+            'mime-type': 'application/json',
+            data: expect.any(Object),
+            lastmod_time: undefined,
+            byte_count: undefined,
+          },
+        ],
+        '~thread': {
+          thid: expect.any(String),
+          pthid: undefined,
+          sender_order: undefined,
+          received_orders: undefined,
+        },
+        '~please_ack': { on: ['RECEIPT'] },
+        '~service': undefined,
+        '~attach': undefined,
+        '~timing': undefined,
+        '~transport': undefined,
+        '~l10n': undefined,
+      })
+    } else {
+      throw new Error('Missing Connection Id')
+    }
```

Comment:
Can we invert these again? Then we don't need the else clause and we can keep most code top level